git_machete/cmd.py

```diff
@@ -1749,14 +2493,8 @@ def is_merged_to(cli_ctxt: CommandLineContext, b: str, target: str) -> bool:
         return contains_equivalent_tree(cli_ctxt, b, target)
 
 
-def is_merged_to_upstream(cli_ctxt: CommandLineContext, b: str) -> bool:
-    if b not in up_branch:
-        return False
-    return is_merged_to(cli_ctxt, b, up_branch[b])
-
-
-def infer_upstream(cli_ctxt: CommandLineContext, b: str, condition: Callable[[str], bool] = lambda u: True, reject_reason_message: str = "") -> Optional[str]:
-    for sha, containing_branch_defs in match_log_to_filtered_reflogs(cli_ctxt, b):
+def infer_upstream(machete_context: MacheteClient, cli_ctxt: CommandLineContext, b: str, condition: Callable[[str], bool] = lambda u: True, reject_reason_message: str = "") -> Optional[str]:
```

Comment:
Also definitely a part of `MacheteClient`.

Reply:
Also! pls hunt down the remaining occurrences of `machete_context` (still 7 left)